From a3f63fada554ee20c5d4e0b282f7212c57d5454c Mon Sep 17 00:00:00 2001
From: Pavel Kalvoda <me@pavelkalvoda.com>
Date: Sun, 22 Mar 2020 12:02:11 +0100
Subject: [PATCH] Shared lib versioning fix

---
 src/CMakeLists.txt | 14 +++++++++++---
 1 file changed, 11 insertions(+), 3 deletions(-)
 .
 Ubuntu notes:
 Please see the following for details on this change:
 https://bugs.launchpad.net/ubuntu/+source/libfido2/+bug/1864439/comments/13
 https://github.com/PJK/libcbor/issues/52#issuecomment-602180765

Origin: upstream, https://github.com/PJK/libcbor/pull/131/commits/a3f63fada554ee20c5d4e0b282f7212c57d5454c
Bug: https://github.com/PJK/libcbor/issues/130
Last-Update: 2020-03-23
diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 953b4f9..00a718e 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -14,12 +14,20 @@ endif()
 add_library(cbor STATIC ${SOURCES})
 add_library(cbor_shared SHARED ${SOURCES})
 
-set_target_properties(cbor_shared PROPERTIES OUTPUT_NAME cbor VERSION ${CBOR_VERSION} SOVERSION ${CBOR_VERSION})
 
-configure_file(libcbor.pc.in libcbor.pc @ONLY)
+if (NOT ${CBOR_VERSION_MAJOR} EQUAL 0)
+	MESSAGE(FATAL_ERROR "Change the shared library version scheme to reflect https://github.com/PJK/libcbor/issues/52.")
+endif()
 
-#http://www.cmake.org/Wiki/CMake:Install_Commands
+set_target_properties(cbor_shared PROPERTIES
+		OUTPUT_NAME cbor
+		VERSION ${CBOR_VERSION}
+		MACHO_COMPATIBILITY_VERSION ${CBOR_VERSION_MAJOR}.${CBOR_VERSION_MINOR}.0
+		SOVERSION ${CBOR_VERSION_MAJOR}.${CBOR_VERSION_MINOR})
+
+configure_file(libcbor.pc.in libcbor.pc @ONLY)
 
+# http://www.cmake.org/Wiki/CMake:Install_Commands
 install(TARGETS cbor_shared
 	ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
 	LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
